packages/axios-extension/src/abap/adt-catalog/services/filestore-service.ts

```diff
@@ -0,0 +1,111 @@
+import { AdtService } from './adt-service';
+import type { AdtCategory, ArchiveFileNode, ArchiveFileNodeType, ArchiveFileContentType } from '../../types';
+import XmlParser from 'fast-xml-parser';
+import type { AdtFileNode } from 'abap/types/adt-internal-types';
+
+/**
+ * FileStoreService implements ADT requests to obtain the content
+ * of deployed archive.
+ */
+export class FileStoreService extends AdtService {
+    /**
+     * @see AdtService.getAdtCatagory()
+     */
+    private static adtCategory = {
+        scheme: 'http://www.sap.com/adt/categories/filestore',
+        term: 'filestore-ui5-bsp'
+    };
+
+    /**
+     * Get ADT scheme ID.
+     *
+     * @returns AdtCategory
+     */
+    public static getAdtCatagory(): AdtCategory {
+        return FileStoreService.adtCategory;
+    }
+
+    /**
+     * If target `path` is a file, the file content is returned as string type.
+     * If target `path` is a folder, files and folders in this folder are returned as an array list
+     * of ArchiveFileNode objects.
+     *
+     * @see ArchiveFileNode
+     * @param type
+     *  Specifies if input `path` refers to a file or a folder. When starting exploring
+     *  the file structure from the root, type should be set to `folder`. The type information
+     *  of files and folders inside root folder can be found in the returned `ArchiveFileNode` entries.
+     * @param appName Deployed Fiori app name
+     * @param path
+     *   Default value is empty string. In this case the output would be folder structure information of the root folder.
+     *   Otherwise provide path to a folder or a file in the deployed archive. E.g. `/webapp/index.html`.
+     * @returns Folder content (ArchiveFileNode[]) | file content (string)
+     */
+    public async getAppArchiveContent<T extends ArchiveFileNodeType>(
+        type: T,
+        appName: string,
+        path = ''
+    ): Promise<ArchiveFileContentType<T>> {
+        const contentType = type === 'folder' ? 'application/atom+xml' : 'application/octet-stream';
+        const config = {
+            headers: {
+                Accept: 'application/xml',
+                'Content-Type': contentType
+            }
+        };
+        const encodedFullPath = encodeURIComponent(`${appName}${path}`);
```

Comment:
If `path` is supplied, does it need to start with a `/`. Does it make sense to check that? May lead to a confusing error message otherwise

Reply:
Updated to detect if path is not starting with '/' and throws error.